### DISCUSSION TITLE
Upgrading on NodeJS

### DISCUSSION THREAD

**Discussion by iCSpotRun** [mKeRix/room-assistant#918] [Q&A]
Hi there. I can’t seem to find how to stop room-assistant on NodeJS. The instructions for upgrading at the following link do not include the command to do so.

https://www.room-assistant.io/guide/upgrading.html#upgrade-process

Thanks in advance for helping this noob out.

**Comment by PeteBa** [comment]
If you did a standard install then you can run `sudo systemctl stop room-assistant`. You can then run `sudo npm i --global --unsafe-perm room-assistant` to download the latest version. You will need to start up the new version with   `sudo systemctl start room-assistant`.

**Comment by iCSpotRun** [comment]
Thank you so much. I was trying to use some variant of 'npm stop'.

Anyway to call the version number via the command line?

**Comment by PeteBa** [comment]
The following will print the current installed version of room-assistant: `npm list --depth=0 | grep room-assistant`.

If you are using 2.19.0 then I suggest setting `instanceBeaconEnabled: false` in the bluetoothLowEnergy section of your config file as this improves stability of the ble adapter.

**Reply by iCSpotRun** [reply]
Thank you again. I am currently running 2.19.0 of a Pi Zero W and BLE is working exceptionally well.

Though BLE from my Home Assistant instance on a Pi4 is quite the opposite. More often than not, it stays at a 'not_home' state even if my device is sitting right next to it for a good period of time.

My BLE settings are:

bluetoothLowEnergy:
  timeout: 60
  allowlist:
    - fc66xxxxxxxx
    - 40D2FE87-xxxx-xxxx-xxxx-xxxxxxxxxxxx
  maxDistance: 3

I'm not particularly concerned given that my full deployment will eventually be more Pi Zero's.

Thanks again for your help!

**Comment by PeteBa** [comment]
No problem. I would suggest  adding `instanceBeaconEnabled: false` to the ble config. This can help improve ble stability on raspberry pis.
